### ISSUE TITLE
Thread parent check

### ISSUE THREAD

**Issue by Glazelf** [discordjs/discord.js#10563]
### Which package is this bug report for?

discord.js

### Issue description

Bot "randomly" (not during boot cycle) crashes with error below:
![Schermafbeelding 2024-10-17 130011](https://github.com/user-attachments/assets/7424bc20-8cdb-46e6-86b1-d43df3ac55f2)

Seems to be an issue with thread parent checks or something. I talked to @almostSouji about this problem [here](https://canary.discord.com/channels/222078108977594368/824411059443204127/1296427687518404648), for some context.

### Versions

- discord.js 14.16.3

### Issue priority

Low (slightly annoying)

### Which partials do you have configured?

User, Channel, GuildMember, Message, Reaction

### Which gateway intents are you subscribing to?

Guilds, GuildMembers, GuildModeration, GuildMessages, GuildMessageReactions, DirectMessages, MessageContent

**Comment by almostSouji** [comment]
Seems to come from here:
https://github.com/discordjs/discord.js/blob/0873f9a4c3a817ce9cf3821f713a871bc3902eb3/packages/discord.js/src/util/Channels.js#L69-L73

Which would imply a channel that does not have `.threads` (or another value) is being set as a parent (since the `.parent` seems to exist) at some point in time.

**Comment by almostSouji** [comment]
Looking at the partials, could this potentially be caused by the parent being partial?
So we receive the partial parent and populate the cache with it; then the thread is being constructed and the partial parent does not have the `.threads`?

**Comment by Glazelf** [comment]
Encountered this a second time earlier today. Error below, on D.JS 14.17.2.
![chrome_g16Jna4TAn](https://github.com/user-attachments/assets/0c97c3c2-1ffd-400e-8fe2-8749f4d83088)

**Comment by Jiralite** [comment]
Do me a favour and install `14.17.3-dev.1736101599-df68b469f`. I added some debug logs regarding as soon as you receive incoming data and what happens right before that crash. It's running 14.17.2 with these changes: https://github.com/discordjs/discord.js/compare/v14...14.7.2-1.

Listen to debug logs like so:

```TypeScript
import { Events } from "discord.js";

client.on(Events.Debug, (message) => {
	// Optionally filter for the specific debug logs.
	// if (!message.startsWith("[14.7.2 DEBUG]")) return;
	console.log(message);
});
```

Yes, it does say 14.7.2 instead of 14.17.2, but don't worry about that! It's just a string.

I'm curious to see what the incoming data is, and what it eventually resolved to, as well as what `parent` actually is, etc. Might help point us in the right direction, especially if this is not reproducible.

**Comment by Glazelf** [comment]
Installed the version you specified, bot crashes on boot with the following error: 
![image](https://github.com/user-attachments/assets/e0740e9a-3caf-4837-ae2c-f61deb10d60d)

**Comment by Jiralite** [comment]
Well then. That sounds like a bug in itself... hold off on that, I guess (or implement the changes yourself, if you know how to patch versions).

**Comment by Glazelf** [comment]
Haha I have no idea. But this is a bug with djs then right, not with my codebase? 
Hope this helps someone more wellread into the djs codebase fix this, then! :)

**Comment by Jiralite** [comment]
> Haha I have no idea. But this is a bug with djs then right, not with my codebase?

Cannot answer that without knowing what's going on first.

Try using 14.17.3-dev.1736104404-15cbca941... should not get that error any more.

**Comment by Glazelf** [comment]
14.17.3-dev.1736104404-15cbca941 boots succesfully and does not seem to produce any errors from what I can tell.
Though as said I couldn't really seem to repro the crash on-demand so I don't know if that is fixed.

**Comment by Jiralite** [comment]
Listen to the debug event and when you crash, share the logs!

**Comment by Glazelf** [comment]
Will do, however I did go 3 months without encountering this error before today so I'm not sure how long that will take :P

**Comment by Glazelf** [comment]
Hi Jiralite, are the changes you made for 14.17.3-dev.1736104404-15cbca941 also included in the stable 14.17.3 release?

**Comment by Jiralite** [comment]
No. Anyway, The patches in 14.17.3 are already in 14.17.3-dev.1736104404-15cbca941, so you're not missing anything.

**Comment by Glazelf** [comment]
This seems to be the debug logging I got corresponding to this crash. Hope this helps. The logs below this are seemingly the bot caching a bunch of guilds.
I tried to get the logs from the console but since this happened pretty early tonight I can't scroll that far up due to all the logging that's happened since

![Image](https://github.com/user-attachments/assets/ed3f6b6f-b927-4ac8-ad77-d526f22ed586)

**Comment by Glazelf** [comment]
The following doesn't seem to log anything, which I think is strange. 
Might indicate an interaction happening while the bot is being removed from a guild or something? 
```js
let problemChannel = await interaction.client.channels.fetch("581776008978497537");
console.log(problemChannel);
```

**Comment by almostSouji** [comment]
Do you get anything with the raw api call?
```js
interaction.client.rest.get("/channels/581776008978497537")
```

**Comment by Glazelf** [comment]
Seems to have the same result.
This is the exact code I tried:
```js
let problemChannel = await interaction.client.rest.get("/channels/581776008978497537");
console.log(problemChannel);
```

**Comment by almostSouji** [comment]
Do you get anything from curl / a rest client of your choosing (postman, insomnia, etc.)?

**Comment by Glazelf** [comment]
Sent a GET request to https://discord.com/api/v10/channels/581776008978497537
Received:
```json
{
    "message": "Missing Access",
    "code": 50001
}
```
Other channels work fine.

**Comment by almostSouji** [comment]
i don't see how you'd get a thread but not the channel the thread is for, since the parent should implicitly deny your ability to get the thread

**Comment by Glazelf** [comment]
I was thinking it might be some leftover caching or interaction that resolves shortly after the bot is removed from a guild but I haven't tested this nor do I know if it even passes the smell test.
Otherwise I have no idea how this happens either.

**Comment by didinele** [comment]
Are you sure you used the right bot token in your cURL/whatever client you used?

**Comment by Glazelf** [comment]
Yes, I tested other channels and those worked. This is the only bot that has access to the channel shown below.

![Image](https://github.com/user-attachments/assets/d511733f-3368-4a7f-863d-5aacc909ff31)

**Comment by Glazelf** [comment]
Occurred in october which lead me to make this issue.
Occurred again january 5th, which lead me to run a build that should have a bit more debug logging in regards to this thanks to Jiralite.
And it occurred early last night.
Which is three times total in a span of roughly six months. 
I have to assume there is some uncaught interaction between systems somewhere that causes this behaviour but I have no idea how to reproduce this.

**Comment by Qjuh** [comment]
You omitted most of the 14.17.2 debug log on that screenshot of your webhook logs. Could you copy/show the full log please?

**Comment by Glazelf** [comment]
Discussed the above questions on Discord.
Shared a log with Qjuh but it seems to be incomplete since I'm storing them to embeds which are limited to 6000 characters.
Fetching issues were unrelated, fetching the channel listed in the initial log screenshot returns 403 error as intended when fetched properly.
Will revisit how I'm storing these logs for permanence, otherwise unless someone has ideas I guess I'll revive this issue in another three months when this issue reoccurs.

**Comment by almostSouji** [comment]
it may even make sense to (although i hate suggesting this) make an edit in the djs code to log the `.parent` in that code section, if it does not have threads. then we'd at least know what the "something" is i mentioned above
